Index: SWP/src/main/java/com/backendVn/SWP/repositories/InvoiceRepository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.backendVn.SWP.repositories;\r\n\r\nimport com.backendVn.SWP.entities.Invoice;\r\nimport org.springframework.data.jpa.repository.JpaRepository;\r\n\r\npublic interface InvoiceRepository extends JpaRepository<Invoice, Integer> {\r\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SWP/src/main/java/com/backendVn/SWP/repositories/InvoiceRepository.java b/SWP/src/main/java/com/backendVn/SWP/repositories/InvoiceRepository.java
--- a/SWP/src/main/java/com/backendVn/SWP/repositories/InvoiceRepository.java	
+++ b/SWP/src/main/java/com/backendVn/SWP/repositories/InvoiceRepository.java	
@@ -4,4 +4,5 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface InvoiceRepository extends JpaRepository<Invoice, Integer> {
+    Invoice findByRequestID(Integer requestID);
 }
\ No newline at end of file
